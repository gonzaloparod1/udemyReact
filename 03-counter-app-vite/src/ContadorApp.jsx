import { useState } from 'react';

function Contador() {
  const [contador, setContador] = useState(0); // Declaramos un estado inicial con valor 0

  return (
    <div>
      <p>Has hecho clic {contador} veces</p>
      <button onClick={() => setContador(contador + 1)}>
        Incrementar
      </button>
    </div>
  );
}

export default Contador;