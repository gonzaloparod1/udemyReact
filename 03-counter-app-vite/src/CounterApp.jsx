import PropTypes from 'prop-types';

export const CounterApp = ({value}) => {

return (
<>
    <h1>CounterApp</h1>
    {/* <h2>{(newMessage.title)}</h2> */}
    {/* {<code>{JSON.stringify (newMessage)}</code>} */}
    <p>{value}</p>
</>
)
}   

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}