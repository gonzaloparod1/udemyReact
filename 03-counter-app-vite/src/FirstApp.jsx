/*import { Fragment } from 'react';

export const FirstApp = () => {
return (
    <Fragment><h1>Gonzalo</h1>
    <p>Soy un subtitulo</p></Fragment>
    )
}
*/

/*
const newMessage = [1,2,3,4,5,6,7,8,9];

export const FirstApp = () => {

return (
    <><h1>{ newMessage }</h1>
    <p>Soy un subtitulo</p></>
    )
}
*/

// const newMessage = {
//     message: 'Hola Mundo',
//     title: 'soy Gonzalo' 
// };


// const getResult = () => {
//     return a + b;
// }

    
    // console.log(props);

    // if (!title){
    //     throw new Error('El title no existe');
    // }
    
    import PropTypes from 'prop-types';

    export const FirstApp = ({title, subtitle, name}) => {

    return (
    <>
        <h1>CounterApp</h1>
        {/* <h2>{(newMessage.title)}</h2> */}
        {/* {<code>{JSON.stringify (newMessage)}</code>} */}
        <p> {value} </p>

    </>
    )
}


FirstApp.propTypes = {
    value: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
}

FirstApp.defaultProps = {
    name: 'Gonzalo Parodi',
    subtitle: 'No hay subtitulo',
    title: 'No hay titulo',
}