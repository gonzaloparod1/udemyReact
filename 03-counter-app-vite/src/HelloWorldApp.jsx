/*
export function App(){
    / document.createElement...
    return (<h1>Hola Mundo!!!</h1>);
}
*/

// export const HelloWorldApp = () => {
//   return (
//     <div>Hello World App</div>
//   )
// }

// export const HelloWorldApp = () => <h1>Arrow Function</h1>


export const HelloWorldApp = () => {
    
    return (
    <h1>Hola Mundo!!!</h1>);
}


// FirstApp