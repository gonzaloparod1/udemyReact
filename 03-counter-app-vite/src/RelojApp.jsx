import { useState, useEffect } from 'react';

function Reloj() {
  const [hora, setHora] = useState(new Date().toLocaleTimeString());

  useEffect(() => {
    const intervalo = setInterval(() => {
      setHora(new Date().toLocaleTimeString());
    }, 1000);

    // Limpiar el intervalo al desmontar el componente
    return () => clearInterval(intervalo);
  }, []); // [] asegura que el efecto se ejecute solo una vez al montar

  return <p>Hora actual: {hora}</p>;
}

export default Reloj;
