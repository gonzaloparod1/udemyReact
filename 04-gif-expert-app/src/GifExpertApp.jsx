export const GifExpertApp = () => {
    return (
        <>
        <h1>GifExpert App</h1>
        </>
    )
}