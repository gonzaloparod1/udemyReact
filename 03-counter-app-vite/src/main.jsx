import React from 'react';
import ReactDOM from 'react-dom/client';
// import { HelloWorldApp } from './HelloWorldApp';
// import { FirstApp } from './FirstApp';
// import ContadorApp from './ContadorApp';
// import Reloj from './RelojApp';
import { CounterApp } from './CounterApp';
// import {App} from './HelloWorldApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={123}/>
    </React.StrictMode>
)