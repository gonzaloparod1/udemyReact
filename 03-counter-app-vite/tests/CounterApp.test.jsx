import {fireEvent, render, screen} from '@testing-library/react';
import {CounterApp} from '../src/CounterApp';


describe('Pruebas en el <CounterApp />', () => {
    
    const initialValue = 10;
    
    test ('debe de hacer match con el snapshot', () => {
        
        const {containder} = render(<CounterApp value={initialValue}/>); 
        expect (containder).toMatchSnapshot();
    });

    test ('debe de mostrar el valor por inicial de 100 <Counter App value={100}>', () => {
        
        render(<CounterApp value={100}/>); 
        expect (screen.getByText('-1')).toBeTruthy();
        // expect(screen.getByRole('heading',{level: 2}).innerHTML).toContain('100');
    });

    test('debe de incrementar con el boton +1',()=>{
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('+1')).toBeTruthy();
    })

    test('debe de incrementar con el boton -1',()=>{
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByText('-1'));
        screen.debug()
        expect(screen.getByText('-1')).toBeTruthy();
    })

    test('debe de funcionarel boton de reset',()=>{
        render(<CounterApp value={355}/>);
        fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}));
        // fireEvent.click(screen.getByText('Reset'));
        // expect(screen.getByText(355)).toBeTruthy();
    })

}); 