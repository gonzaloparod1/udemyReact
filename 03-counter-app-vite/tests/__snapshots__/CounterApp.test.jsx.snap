// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp /> debe de hacer match con el snapshot 1`] = `undefined`;

exports[`Pruebas en el <CounterApp /> debe de hacer match conel snapshot 1`] = `undefined`;
